src/bootstrap/src/core/config/config.rs

```diff
@@ -1621,6 +1623,12 @@ impl Config {
         // Verbose flag is a good default for `rust.verbose-tests`.
         config.verbose_tests = config.is_verbose();
 
+        if let Some(jobs) = jobs {
+            if jobs > 0 {
+                config.jobs = Some(jobs);
+            }
+        }
```

Comment:
We should handle it here: https://github.com/rust-lang/rust/blob/e057ca519e5c557b8280efc7fca54d66199324df/src/bootstrap/src/core/config/config.rs#L1293

Reply:
Note that there's two layers of precedence, the line there is the top priority `--jobs` flag. I changed the logic to better reflect the precedence such that:

- `config.jobs` and `flags.jobs` are initially `None`.
- `config.jobs` is `Some` based on `--jobs=N` if `N > 0` and still `None` otherwise.
- If `config.jobs` is still `None`, then we fallback to `--set build.jobs` then `[build] jobs`. Note that the `build` toml section is read below this.

Reply:
(Remark: I found this config layering somewhat hard to follow)